pkg/internal/apis/handlers/clone_record.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: 2025 Forkbomb BV
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+package handlers
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/forkbombeu/credimi/pkg/internal/apierror"
+	"github.com/forkbombeu/credimi/pkg/internal/canonify"
+	"github.com/forkbombeu/credimi/pkg/internal/middlewares"
+	"github.com/forkbombeu/credimi/pkg/internal/routing"
+	"github.com/pocketbase/pocketbase/apis"
+	"github.com/pocketbase/pocketbase/core"
+	"github.com/pocketbase/pocketbase/tools/hook"
+)
+
+var CloneRecord routing.RouteGroup = routing.RouteGroup{
+	BaseURL:                "/api",
+	AuthenticationRequired: false,
+	Middlewares: []*hook.Handler[*core.RequestEvent]{
+		{Func: middlewares.ErrorHandlingMiddleware},
+	},
+	Routes: []routing.RouteDefinition{
+		{
+			Method:  http.MethodGet,
+			Path:    "/clone-record",
+			Handler: HandleGetCloneRecord,
+		},
+	},
+}
+
+var rng = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func HandleGetCloneRecord() func(*core.RequestEvent) error {
+	return func(e *core.RequestEvent) error {
+		recordID := e.Request.URL.Query().Get("id")
+
+		if recordID == "" {
+			return apis.NewBadRequestError(
+				"Parameter 'id' is required",
+				nil,
+			)
+		}
+
+		originalRecord, err := canonify.Resolve(e.App, recordID)
+		if err != nil {
+			return apierror.New(
+				http.StatusNotFound,
+				"resolve",
+				"failed to resolve collection path",
+				err.Error(),
+			).JSON(e)
+		}
```

Comment:
this should be a general API that can work also for records that do not support canonization
this is why, in the previous comment, I said to use both `collection_name` and `record_id` to find the record